### PR TITLE
feat: cap transaction history

### PR DESCRIPTION
**Pull request by ozwaldorf**
## Why?

Adds cap transaction history

## How?

- [x] Utilize the new cap-sdk `insert_sync` and `archive`/`from_archive`, which guarantees the finality and order of transactions. In the background, the sdk stores a buffer of failed insertions and flushes the buffer using the new batch insertion method 
- [x] upgrade cap submodule
- [x] deploy cap if CAP_ID is not set
- [x] init canister with CAP_ID
- [x] integrate cap-js with the tests to restore txn testing

## Contribution checklist?

- [x] The commit messages are detailed
- [x] It does not break existing features (unless required)
- [x] I have performed a self-review of my own code
- [x] Documentation has been updated to reflect the changes
- [x] Tests have been added or updated to reflect the changes
- [x] All code formatting pass
- [x] All lints pass
- [x] All tests pass

## Security checklist?

- [x] Injection has been prevented (parameterized queries, no eval or system calls)
- [x] Sensitive data has been identified and is being protected properly

### CODE REVIEW

**Comment by scott-dn** [comment]
No test anymore? 😭

**Comment by ozwaldorf** [comment]
(fixed error in commit message)

File: .github/workflows/pr-integration-test-runner.yml [hunk 1]
```diff
@@ -48,6 +48,10 @@ jobs:
           override: true
           target: wasm32-unknown-unknown
 
+      - name: Install ic-cdk-optimizer
```
Comment (scott-dn) [inline .github/workflows/pr-integration-test-runner.yml:51]: Duplicate with the line below.

File: test/package.json [hunk 2]
```diff
@@ -40,6 +41,7 @@
     "@dfinity/candid": "^0.11.1",
     "@dfinity/identity": "^0.11.1",
     "@dfinity/principal": "^0.11.1",
```
Comment (scott-dn) [inline test/package.json:43]: Is it because of cap-js?

Comment (ozwaldorf) [reply test/package.json:43]: Yeah, apparently dfinity changed the principal _arr again between 0.11 and 0.10 😭

File: .github/workflows/pr-integration-test-runner.yml (hunk 1 above)
Comment (ozwaldorf) [reply .github/workflows/pr-integration-test-runner.yml:51]: Yeah, without it, it fails bc of cap's custom build scripts

https://github.com/Psychedelic/DIP721/runs/6732645317?check_suite_focus=true
```
-snip-

   Compiling ic-kit v0.4.3
   Compiling cap-common v0.2.0 (/home/runner/work/DIP721/DIP721/cap/common)
   Compiling ic-history-bucket v0.1.0 (/home/runner/work/DIP721/DIP721/cap/canisters/bucket)
    Finished release [optimized] target(s) in 1m 38s
/bin/sh: 1: ic-cdk-optimizer: not found
node:child_process:902
    throw err;
    ^
Error: Command failed: ic-cdk-optimizer target/wasm32-unknown-unknown/release/ic_history_bucket.wasm -o target/wasm32-unknown-unknown/release/ic_history_bucket-rel-opt.wasm
/bin/sh: 1: ic-cdk-optimizer: not found

-snip- 
```

Comment (scott-dn) [reply .github/workflows/pr-integration-test-runner.yml:51]: I mean, we could delete lines 55-57.

**Comment by scott-dn** [comment]
It seems like the test failed due to lacking npm package.

**Review by scott-dn — APPROVED** [review]

**Pull request merged**